engine/src/eventqueue.cpp

```diff
@@ -1226,68 +1227,92 @@ static void handle_touch(MCStack *p_stack, MCEventTouchPhase p_phase, uint32_t p
 	MCPoint t_touch_loc;
 	t_touch_loc = p_stack->view_viewtostackloc(MCPointMake(x, y));
 	
-	if (p_phase != kMCEventTouchPhaseBegan)
-	{
-		for(t_touch = s_touches; t_touch != nil; t_previous_touch = t_touch, t_touch = t_touch -> next)
-			if (t_touch -> id == p_id)
-				break;
-		
-		if (t_touch != nil)
-		{
-			t_target = t_touch -> target;
-				
-			// MW-2011-09-05: [[ Bug 9683 ]] Make sure we remove (and delete the touch) here if
-			//   it is 'end' or 'cancelled' so that a cleartouches inside an invoked handler
-			//   doesn't cause a crash.			
-			if (p_phase == kMCEventTouchPhaseEnded || p_phase == kMCEventTouchPhaseCancelled)
-			{
-				if (t_previous_touch == nil)
-					s_touches = t_touch -> next;
-				else
-					t_previous_touch -> next = t_touch -> next;
-				
-				delete t_touch;
-			}
-		}
-	}
-	else
-	{
-		t_touch = new (nothrow) MCTouch;
-		t_touch -> next = s_touches;
-		t_touch -> id = p_id;
-		
-		t_target = p_stack -> getcurcard() -> hittest(t_touch_loc.x, t_touch_loc.y);
-		t_touch -> target = t_target -> GetHandle();
-		
-		s_touches = t_touch;
-	}
 
-	if (t_target != nil)
-	{
-        // Touches on widgets are handled differently
-        if (t_target->gettype() == CT_WIDGET)
+    if (p_phase != kMCEventTouchPhaseBegan)
+    {
+        for(t_touch = s_touches; t_touch != nil; t_previous_touch = t_touch, t_touch = t_touch -> next)
```

Comment:
@runrevmark there seems to be a rather large whitespace diff here. Do you want to drop that?